src/main/java/com/paneedah/weaponlib/WeaponSpawnEntity.java

```diff
@@ -82,12 +88,12 @@ protected void onImpact(final RayTraceResult position) {
             Explosion.createServerSideExplosion(weapon.getModContext(), world, this.getThrower(), this, position.hitVec.x, position.hitVec.y, position.hitVec.z, explosionRadius, false, true, isDestroyingBlocks, explosionParticleAgeCoefficient, smokeParticleAgeCoefficient, explosionParticleScaleCoefficient, smokeParticleScaleCoefficient, weapon.getModContext().getRegisteredTexture(explosionParticleTextureId), weapon.getModContext().getRegisteredTexture(smokeParticleTextureId), weapon.getModContext().getExplosionSound());
         } else if (position.entityHit != null) {
             if (this.getThrower() != null)
-                position.entityHit.attackEntityFrom(DamageSource.causeThrownDamage(this, this.getThrower()), damage);
+                position.entityHit.attackEntityFrom(new ProjectileDamageSource(this, this.getThrower()), damage);
```

Comment:
Please stop making the damage sourse a vanilla arrow. That's not correct.

Reply:
The source of damage is not an arrow, it's just that the type of death message is an arrow